packages/shared/providers.yaml

```diff
@@ -4634,18 +4635,36 @@ quickbooks:
     auth_mode: OAUTH2
     authorization_url: https://appcenter.intuit.com/connect/oauth2
     token_url: https://oauth.platform.intuit.com/oauth2/v1/tokens/bearer
-    redirect_uri_metadata:
-        - realmId
     proxy:
+        connection_config:
+            realmId: ${connectionConfig.realmId}
         base_url: https://quickbooks.api.intuit.com
     docs: https://docs.nango.dev/integrations/all/quickbooks
+    connection_config:
+        realmId:
+            type: string
+            title: Quickbooks Realm ID
+            optional: true
+            description: The realmId of your quickbooks account
+            pattern: '^\d{16}$'
+            example: '9341453474484455'
 
 quickbooks-sandbox:
     alias: quickbooks
     display_name: Quickbooks (sandbox)
     proxy:
+        connection_config:
+            realmId: ${connectionConfig.realmId}
```

Comment:
I think it's a follow-up question from the aws PR I didn't get, How are we supposed to use it if it's not in a URL or headers?

Reply:
For the AWS-IAM, since we didn't support the AWS V4 signature authentication mode, we needed a way to prompt the user for the region. This region is then used to generate the signature within the script. For QuickBooks, the `realmId` is required to build the proxy `base_url`. However, after discussing with Khaliq, we decided it was best not to alter the `base_url` as doing so might break compatibility for other users.

Reply:
alright 👌🏻